flink-streaming-java/src/test/java/org/apache/flink/streaming/api/operators/async/queue/UnorderedStreamElementQueueTest.java

```diff
@@ -18,174 +18,108 @@
+		Assert.assertEquals(8, queue.size());
+		Assert.assertFalse(queue.isEmpty());
 
-		CompletableFuture<AsyncResult> secondPoll = CompletableFuture.supplyAsync(
-			() -> {
-				try {
-					return queue.poll();
-				} catch (InterruptedException e) {
-					throw new CompletionException(e);
-				}
-			},
-			executor);
+		record2.complete(Arrays.asList(12));
 
-		record6.complete(Collections.<Integer>emptyList());
-		record4.complete(Collections.<Integer>emptyList());
+		Assert.assertEquals(Arrays.asList(new StreamRecord<>(12, 1L)), popCompleted(queue));
+		Assert.assertEquals(7, queue.size());
+		Assert.assertFalse(queue.isEmpty());
 
-		Thread.sleep(10L);
+		// Should not be completed because R1 has not been completed yet
+		record6.complete(Arrays.asList(16));
+		record4.complete(Arrays.asList(14));
 
-		// The future should not be completed because R1 has not been completed yet
-		Assert.assertFalse(secondPoll.isDone());
-
-		record1.complete(Collections.<Integer>emptyList());
-
-		Assert.assertEquals(record1, secondPoll.get());
+		Assert.assertEquals(Collections.emptyList(), popCompleted(queue));
+		Assert.assertEquals(7, queue.size());
+		Assert.assertFalse(queue.isEmpty());
 
 		// Now W1, R3, R4 and W2 are completed and should be pollable
-		Assert.assertEquals(watermark1, queue.poll());
-
-		// The order of R3 and R4 is not specified
-		Set<AsyncResult> expected = new HashSet<>(2);
-		expected.add(record3);
-		expected.add(record4);
-
-		Set<AsyncResult> actual = new HashSet<>(2);
-
-		actual.add(queue.poll());
-		actual.add(queue.poll());
-
-		Assert.assertEquals(expected, actual);
-
-		Assert.assertEquals(watermark2, queue.poll());
-
-		// since R6 has been completed before and W2 has been consumed, we should be able to poll
-		// this record as well
-		Assert.assertEquals(record6, queue.poll());
+		record1.complete(Arrays.asList(11));
+
+		Assert.assertEquals(Arrays.asList(
+				new StreamRecord<>(11, 0L),
+				new Watermark(2L),
+				new StreamRecord<>(13, 3L),
+				new StreamRecord<>(14, 4L),
+				new Watermark(5L),
+				new StreamRecord<>(16, 7L)),
+				popCompleted(queue));
+		Assert.assertEquals(1, queue.size());
+		Assert.assertFalse(queue.isEmpty());
 
 		// only R5 left in the queue
-		Assert.assertTrue(1 == queue.size());
+		record5.complete(Arrays.asList(15));
 
-		CompletableFuture<AsyncResult> thirdPoll = CompletableFuture.supplyAsync(
-			() -> {
-				try {
-					return queue.poll();
-				} catch (InterruptedException e) {
-					throw new CompletionException(e);
-				}
-			},
-			executor);
+		Assert.assertEquals(Arrays.asList(new StreamRecord<>(15, 6L)), popCompleted(queue));
+		Assert.assertEquals(0, queue.size());
+		Assert.assertTrue(queue.isEmpty());
+		Assert.assertEquals(Arrays.asList(), popCompleted(queue));
+	}
 
-		Thread.sleep(10L);
 
-		Assert.assertFalse(thirdPoll.isDone());
+	/**
+	 * Tests two adjacent watermarks can be processed successfully.
+	 */
+	@Test
+	public void testWatermarkOnly() {
```

Comment:
This test is totally the same with `OrderedStreamElementQueueTest#testWatermarkOnly`, so we could migrate it into `StreamElementQueueTest` instead.